### DISCUSSION TITLE
How to get the result from an object builder

### DISCUSSION THREAD

**Discussion by Bonscho** [microsoft/typespec#3353] [Q&A] [answered]
Guys, I am new to all this. I just implemented my first custom emitter which is a `TypeEmitter<object>`.  Each of the relevant methods return something like `this.emitter.result.declaration(…)`. I think so far I am aligned with the docs. I am wondering, how I can get the result (`Declaration<object>`) from the `AssetsEmitter` in the lib. 
Would be super awesome if someone could point me to the right direction. Maybe I did not get the concept completely …

**Comment by bterlson** [comment, marked as answer]
All of the declarations are pushed into a scope depending on which scope is set in the emitter context. Usually this is just your source file's global scope unless you're doing anything fancy. So check your source file's `globalScope.declarations` array.

The docs aren't super clear here, sorry. You could potentially find [this example](https://github.com/microsoft/typespec/blob/main/packages/compiler/test/emitter-framework/typescript-emitter.ts#L266) from a test case helpful though.

**Reply by Bonscho** [reply]
Thanks a ton for your support. Pointing me to the example brought up the needed details. The docs need some love anyhow …
